fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/api/SavingsAccountTransactionsApiResource.java

```diff
@@ -137,6 +141,20 @@ public String searchTransactions(@PathParam("savingsId") @Parameter(description
         return toApiJsonSerializer.serialize(transactionsData);
     }
 
+    @POST
+    @Path("query")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Search Savings Account Transactions")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = List.class))) })
+    public String advancedQuery(@PathParam("savingsId") @Parameter(description = "savingsId") final Long savingsId,
+            PagedRequest<AdvancedQueryRequest> queryRequest, @Context final UriInfo uriInfo) {
+        final org.springframework.data.domain.Page<JsonObject> result = transactionsSearchServiceImpl.queryAdvanced(savingsId,
+                queryRequest);
+        return this.toApiJsonSerializer.serializePretty(ApiParameterHelper.prettyPrint(uriInfo.getQueryParameters()), result);
```

Comment:
As I said for the datatables API, you can go with Page<...> since this is a new API.

Reply:
The rule is

for V2 it is mandatory to be object response
for V1 new API it is okay to be object response